quinn-proto/src/connection/mod.rs

```diff
@@ -3760,6 +3874,8 @@ pub enum Event {
     DatagramReceived,
     /// One or more application datagrams have been sent after blocking
     DatagramsUnblocked,
+    /// Received an observation of our external address from the peer.
```

Comment:
Explicitly document whether this is necessarily different from the previous one, and the required configuration and peer support.